search_service/api/table.py

```diff
@@ -116,3 +116,52 @@ def get(self, *, field_name: str, field_value: str) -> Iterable[Any]:
 
             err_msg = 'Exception encountered while processing search request'
             return {'message': err_msg}, HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+class SearchTableFilterAPI(Resource):
+    """
+    Search Table API using search filtering
+
+    This API should be generic enough to support every search filter use case.
+    TODO: Deprecate the SearchTableFieldAPI for this more flexible API
+    """
+    def __init__(self) -> None:
+        self.proxy = get_proxy_client()
+
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('index', required=False, default=TABLE_INDEX, type=str)
```

Comment:
I will let Tao comment I'm unsure of the implications given this endpoint is support to search tables.

Reply:
Given this resource file is for table, let's still use the table index. Once we introduces more index(user, dashboard, metric etc), it will be hard to use single default index in the config.